client/rest-high-level/src/test/java/org/elasticsearch/client/RequestTests.java

```diff
@@ -307,4 +305,45 @@ public void testEndpoint() {
         assertEquals("/a/b/_create", Request.endpoint("a", "b", "_create"));
         assertEquals("/a/b/c/_create", Request.endpoint("a", "b", "c", "_create"));
     }
-}
\ No newline at end of file
+
+    private void enrichReplicationRequest(ReplicatedWriteRequest request, Map<String, String> expectedParams) {
+        if (randomBoolean()) {
+            String timeout = randomTimeValue();
+            request.timeout(timeout);
+            expectedParams.put("timeout", timeout);
+        } else {
+            expectedParams.put("timeout", ReplicationRequest.DEFAULT_TIMEOUT.getStringRep());
+        }
+
+        if (randomBoolean()) {
+            WriteRequest.RefreshPolicy refreshPolicy = randomFrom(WriteRequest.RefreshPolicy.values());
+            request.setRefreshPolicy(refreshPolicy);
+            if (refreshPolicy != WriteRequest.RefreshPolicy.NONE) {
+                expectedParams.put("refresh", refreshPolicy.getValue());
+            }
+        }
+    }
+
+    private void enrichDocWriteRequest(DocWriteRequest request, Map<String, String> expectedParams) {
+        // There is some logic around _create endpoint and version/version type
+        if (request.opType() == DocWriteRequest.OpType.CREATE) {
```

Comment:
This part should be in the testIndex() method because it is specific to Index requests, it could be:

```
public void testIndex()  {
    ...
    if (request.opType() == DocWriteRequest.OpType.CREATE) {
        ...
    } else {
        setRandomVersion(indexRequest, expectedParams);
        setRandomVersionType(indexRequest, expectedParams);
        ...
   }
```